charms/jupyter-controller/src/charm.py

```diff
@@ -59,110 +80,181 @@ def __init__(self, *args):
-                                    "verbs": ["*"],
-                                },
-                                {
-                                    "apiGroups": [""],
-                                    "resources": ["events"],
-                                    "verbs": ["create", "get", "list", "patch", "watch"],
-                                },
-                                {
-                                    "apiGroups": [""],
-                                    "resources": ["pods"],
-                                    "verbs": ["get", "list", "watch"],
-                                },
-                                {"apiGroups": [""], "resources": ["services"], "verbs": ["*"]},
-                                {
-                                    "apiGroups": ["kubeflow.org"],
-                                    "resources": [
-                                        "notebooks",
-                                        "notebooks/finalizers",
-                                        "notebooks/status",
-                                    ],
-                                    "verbs": ["*"],
-                                },
-                                {
-                                    "apiGroups": ["networking.istio.io"],
-                                    "resources": ["virtualservices"],
-                                    "verbs": ["*"],
-                                },
-                            ],
-                        }
-                    ]
-                },
-                "containers": [
-                    {
-                        "name": "jupyter-controller",
-                        "imageDetails": image_details,
-                        "command": ["./manager"],
-                        "envConfig": {
-                            "USE_ISTIO": "true",
-                            "ISTIO_GATEWAY": f"{model}/kubeflow-gateway",
-                            "ENABLE_CULLING": config["enable-culling"],
-                        },
-                    }
-                ],
+        ret_env_vars = {
+            "USE_ISTIO": "true",
+            "ISTIO_GATEWAY": f"{self.model.name}/kubeflow-gateway",
+            "ENABLE_CULLING": config["enable-culling"],
+        }
+
+        return ret_env_vars
+
+    @property
+    def _jupyter_controller_layer(self) -> Layer:
+        """Create and return Pebble framework layer."""
+        env_vars = self.service_environment
+
+        layer_config = {
+            "summary": "jupyter-controller layer",
+            "description": "Pebble config layer for jupyter-controller",
+            "services": {
+                self._container_name: {
+                    "override": "replace",
+                    "summary": "Entrypoint of jupyter-controller image",
+                    "command": self._exec_command,
+                    "startup": "enabled",
+                    "environment": env_vars,
```

Comment:
```suggestion
                    "environment": self.service_environment,
```

just use property directly here

Reply:
@i-chvets any reason why not?

Reply:
Done.